File: 그리디알고리즘/문자열 뒤집기.py
string = list(map(int, input()))

cnt = [0,0] #0집합, 1집합의 수 저장

now = string[0]
cnt[string[0]] = 1

for i in range(1,len(string)):
    if string[i] != now:
        now = string[i]
        cnt[now] += 1
print(min(cnt[0],cnt[1]))

File: JavaScript/그리디/문자열 뒤집기.js
const s = "00011001010100010000";

// 최소 뒤집기 횟수로 모두 같은 숫자로 만들기
// 연속된 0 과 1의 부분집합 개수를 구한다.
// 최솟값을 출력한다.

const s_array = s.split("");

let now = "";
const count = [0, 0];

while (s_array.length) {
  const char = s_array.pop();
  if (now !== char) {
    count[char]++;
    now = char;
  }
}

console.log(Math.min(...count));
